docs/src/consensus/block.md

```diff
@@ -37,24 +37,40 @@ The genesis block is hardcoded in the source code, see `genesis_block` in `block
- - ***f)*** there is an ancestor block `luca` of the current block such that for each uncle block `uncle`
-   - `uncle` is valid
-   - `luca` is an ancestor of `uncle`
-   - neither `luca` nor any of the blocks between `luca` and the current block list `uncle` as an uncle block
- - ***g)*** the `version` matches that of its predecessor or is member of a predefined list of exceptions
- - ***h)*** the `height` is one greater than that of its predecessor
- - ***i)*** the `timestamp` is greater than that of its predecssor
- - ***j)*** the network statistics trackers are updated correctly
- - ***k)*** the variable network parameters are updated correctly.
+A block is incrementally valid, relative to a predecessor block, iff (all of):
+
+ 0. The block's relation to its predecessor is correct, specifically (all of):
+     - a) The block height is that of its predecessor plus one.
+     - b) The `prev_block_digest` of the header equals the hash of the given predecessor block.
+     - c) The `prev_block_digest` is the most recently accumulated element in the `block_mmr_accumulator`.
+     - d) The block timestamp must be later than that of the predecessor plus the minimum block time. The minimum block time is set to 60 seconds.
+     - e) The target difficulty was updated in accordance with the <span style="color:red">difficulty control algorithm</span>.
+     - f) The current block's cumulative proof-of-work number equals that of the predecessor plus the predecessor's difficulty.
+     - g) The block timestamp is not set further in the future, relative to the host machine's clock, than the futuredating limit, which is 5 minutes.
+ 1. The block's stand-alone non-transaction data is correct, specifically (all of):
+     - a) The block appendix contains the expected claims. At present, the list of expected claims contains only one item: the claim that the block's transaction is valid.
+     - b) The block appendix does not contain too many claims. The limit is 500.
+     - c) The block proof is of the right type, namely `SingleProof`. (This variant lives on `BlockProof` and is distinct from the like-named variant living on `TransactionProof`.) The alternatives are `Genesis` and `Invalid`.
+     - d) The proof passes Triton VM verification. The input for the claim is the block mast hash; the output is all appendices; and the program is the `BlockProgram`. The `BlockProgram` merely proves the integral verification of all claims in the appendix.
+     - e) The block does not exceed the maximum size, which is set to 1'000'000 `BFieldElement`s, or 8 MB.
+ 2. The block's transaction is correct, specifically (all of):
+     - a) The removal records can be *unpacked*. Phrased differently, failure to unpack the removal records results in a format error.
+     - b) All removal records must be *removable* from the mutator set as it was after the predecessor block.
+     - c) All removal records must be unique.
```

Comment:
It's the absolute index sets of the removal records that must be unique.

Reply:
My first reaction was, "that's implied." But it's not!

Specifically, you could have distinct removal records that have the same absolute index set. That's disallowed.

So: good catch!